### PR TITLE
Add securitygroup support

### PR DESCRIPTION
**Pull request by notandy**
This PR adds the entity securitygroup to cyclone and supports the one-way syncing of security groups and all related remote security groups.

### CODE REVIEW

File: pkg/security_group.go [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+			} else {
+				// create remote security group if it doesn't exist
+				remoteSecurityGroup, err := migrateSecurityGroup(srcNetworkClient, dstNetworkClient, rule.RemoteGroupID, "")
+				if err != nil {
+					return nil, err
+				}
+				// update rule with new remote group ID
+				remoteGroupID = remoteSecurityGroup.ID
+			}
+		}
```
Comment (kayrus) [inline pkg/security_group.go:181]: is there a chance to rework this with less if/then/else?

**Review by kayrus — COMMENTED** [review]
Thanks for the PR. See my comments.

File: pkg/security_group.go [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+		Description: sg.Description,
+	}
+	newSecurityGroup, err := groups.Create(dstNetworkClient, createOpts).Extract()
+	if err != nil {
+		return nil, err
+	}
+	// store mapping of source and destination security group IDs in case we encounter a rule with a remote group
+	syncedSrcDstSecurityGroups[securityGroup] = newSecurityGroup.ID
+
+	// delete default egress rules to get a clean slate
```
[the remark this answers is not in the document]
Comment (notandy) [reply pkg/security_group.go:155]: hehe, It was no typo :) https://www.dictionary.com/browse/clean-slate

**Comment by notandy** [comment]
amended the PR

File: pkg/security_group.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (kayrus) [reply pkg/security_group.go:155]: oh, good to know :)

File: README.md [hunk 3]
```diff
@@ -35,15 +35,16 @@ Usage:
...
-  server      Clone a server
-  share       Clone a share
-  version     Print version information
-  volume      Clone a volume
+  backup
+  completion    Generate the autocompletion script for the specified shell
+  help          Help about any command
+  image         Clone an image
+  secret        Clone a secret
+  securitygroup Clone a securitygroup
```
Comment (kayrus) [inline README.md:43]:
```suggestion
  security-group Clone a security group
```

**Review by kayrus — APPROVED** [review]
LGTM, thanks

**Pull request merged**